engine/consensus/compliance/core.go

```diff
@@ -93,21 +93,22 @@ func NewCore(
 
 // OnBlockProposal handles incoming block proposals.
 func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.BlockProposal, inBlockRangeResponse bool) error {
+	block := proposal.Block.ToInternal()
```

Comment:
Would make the most sense to me, to only convert once here too (requiring changing the type of the pending block storage here as well)